tools/scuffed-scraper/main.py

```diff
@@ -0,0 +1,612 @@
+# DTU Scuffed Course Scraper
+# Copyright (C) 2023 Sofus Albert Høgsbro Rose
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import sys
+
+SUPPORTED_PY_MAJOR = 3
+SUPPORTED_PY3_VERSIONS = [11, 12, 13, 14]
+if not all(
+	[
+		sys.version_info.major == SUPPORTED_PY_MAJOR,
+		sys.version_info.minor in SUPPORTED_PY3_VERSIONS,
+	],
+):
+	print("Python 3.X minor version must be in:", SUPPORTED_PY3_VERSIONS)
+	sys.exit(1)
+
+import enum
+import json
+import re
+from pathlib import Path
+from urllib.parse import quote as urlquote
+
+import requests
+import tomlkit
```

Comment:
Instead of tomlkit, could we use `toml` as shipped in stdlib, and if pre 3.11 use tomllib. This is what we do for the documentation, so staying consistent with that would be great.

Reply:
Ah yes, well I agree! Issue was `tomllib` doesn't have a `.dumps` function - only `.loads` (https://docs.python.org/3/library/tomllib.html).

I couldn't find `toml` in the stdlib at all; only in PyPi. So, faced with the prospect of having to use a 3rd-party library for TOML writing _anyway_, I figured hey, `tomlkit` looks pretty nice.

Reply:
You are absolutely correct. I had missed that the PEP for adding `tomllib` was intentionally omitting the `dumps` part... :(

As for the naming, I was wrong. As you said, `tomllib` for 3.11, and `toml` for pre 3.11.
Is there a particular reason for `tomlkit` and not `toml`?
Choose any :)

Reply:
No worries aha, and no reason for `tomlkit`, just needed a decently maintained thing.